*** operator.c	Tue Apr  7 15:30:29 2009
--- operator.c	Tue Apr  7 15:27:44 2009
***************
*** 272,279 ****
  		case ZEND_ASSIGN_DIM:
  		{
  			zend_op *opdata = opline + 1;
! 			zval *object = php_operator_zval_ptr(&(opline->op1), &free_obj, execute_data TSRMLS_CC);
! 			zval *prop = php_operator_zval_ptr(&(opline->op2), &free_prop, execute_data TSRMLS_CC);
  
  			if (!object || Z_TYPE_P(object) != IS_OBJECT || !prop) {
  				/* Let orignal handler throw error */
--- 271,278 ----
  		case ZEND_ASSIGN_DIM:
  		{
  			zend_op *opdata = opline + 1;
! 			zval *object = php_operator_zval_ptr(&(opdata->op1), &free_obj, execute_data TSRMLS_CC);
! 			zval *prop = php_operator_zval_ptr(&(opdata->op2), &free_prop, execute_data TSRMLS_CC);
  
  			if (!object || Z_TYPE_P(object) != IS_OBJECT || !prop) {
  				/* Let orignal handler throw error */
